tests/test_execute_command.py

```diff
@@ -0,0 +1,51 @@
+from unittesting import DeferrableTestCase
+import sublime
+from os.path import dirname
+from LSP.plugin.core.types import ClientConfig, ClientStates, LanguageConfig
+from LSP.plugin.core.test_session import MockClient
+from LSP.plugin.core.sessions import Session
+from LSP.plugin.core.registry import windows  # , session_for_view
+from LSP.plugin.core.settings import client_configs
+from LSP.plugin.core.types import Command
+
+test_language = LanguageConfig("test", ["text.plain"], ["Plain text"])
+test_commands = [Command("command1", dict()), Command("command2", dict())]
+text_config = ClientConfig("test", [], None, languages=[test_language], commands=test_commands,)
+
+
+class LspExecuteCommandTests(DeferrableTestCase):
+
+    def setUp(self):
+        self.view = sublime.active_window().new_file()
+        self.old_configs = client_configs.all
+        client_configs.all = [text_config]
+
+    def test_show_inputlistener(self):
+        wm = windows.lookup(self.view.window())
+        wm._configs.all.append(text_config)
+
+        session = Session(text_config, dirname(__file__), MockClient())
+        session.state = ClientStates.READY
+        wm._sessions[text_config.name] = session
+
+        self.view.run_command('lsp_execute')
+
+        # popup should be visible eventually
+        yield self.view.is_popup_visible()
+        # TODO: check the content of the pallet, select one element and hit enter
+
+    def test_execute_command(self):
+        wm = windows.lookup(self.view.window())
+        wm._configs.all.append(text_config)
+
+        session = Session(text_config, dirname(__file__), MockClient())
+        session.state = ClientStates.READY
+        wm._sessions[text_config.name] = session
+        self.view.run_command("lsp_execute_command", {"command_name": "command1"})
```

Comment:
@rwols I am seeing "no config found or enabled for view" in the console when running the test. Do you know why the config is `None`? even though  I do like the other tests  
```
wm = windows.lookup(self.view.window())
wm._configs.all.append(text_config)
```

Reply:
These tests can be a bit painful to work on as they are sensitive to state changes from other tests and your current editing work. I recommend re-opening Sublime and running only 1 test case at a time to keep things sane. Also, if you make implementation changes, reloading package from the sublime palette may be needed.

"no config found or enabled for view" might be a red herring (eg. from `new_file()` in setUp)  - adding additional logging is unfortunately the best advice I can give.
I've never tried starting a debugger on the sublime python scripts, maybe that works for you?

Reply:
I am running this test file alone, so it shouldn't be related to previous state. 
for some reason [score selector](https://github.com/tomv564/LSP/blob/master/plugin/core/configurations.py#L42) is returning 0 even though I open `/testfile.txt` via the view and I register a config for txt files `LanguageConfig("test", ["text.plain"], ["Plain text"])`

Reply:
The command is not invoked your test because Sublime's command naming is a bit tricky. Try
`lsp_execute` instead of `lsp_execute_command`!

Much of the test setup should work like test_hover.py

Reply:
Forgot to add, client responses are hardcoded in MockClient (see test_session.py). You'll want to add:
```'workspace/executeCommand': {}``` to its `responses` dict.

Reply:
@tomv564  I copied the same code as in `test_hover.py` and still `no config found or enabled for view LSP/tests/testfile.txt` :(
I have no clue why `score` in `get_scope_client_config` is zero even though the client config is found. 

Did it work for you ?